includes/multi-currency/FrontendCurrencies.php

```diff
@@ -160,4 +190,34 @@ public function add_currency_to_cart_hash( $hash ): string {
 		$currency = $this->multi_currency->get_selected_currency();
 		return md5( $hash . $currency->get_code() . $currency->get_rate() );
 	}
+
+	/**
+	 * Checks whether currency code used for formatting should be overridden.
+	 *
+	 * @return bool
+	 */
+	public function should_override_currency_code(): bool {
+		return $this->utils->is_call_in_backtrace(
+			[
+				'WC_Shortcode_My_Account::view_order',
+				'WC_Shortcode_Checkout::order_received',
+			]
+		);
+	}
+
+	/**
+	 * Inits order currency code from ID in current URL
+	 *
+	 * @return void
+	 */
+	public function init_order_currency() {
+		if ( null !== $this->order_currency ) {
+			return;
+		}
+
+		$order_id = 126; // todo get order ID from the URL.
```

Comment:
I'll go further with this one if the approach sounds good. So far I have doubts about it because we'll need to get id from the query string, then query the database and get the order. Maybe there is another solution that is simpler and I am not aware of it.

Reply:
Out of date, closing this one.